src/aioswitcher/api/__init__.py

```diff
@@ -103,12 +108,21 @@ def __init__(
         self._device_key = device_key
         self._port = port
         self._connected = False
+        if token:
+            self._token = tok.he(token)  # type: ignore
+        else:
+            self._token = ""
```

Comment:
```suggestion
        if token:
            self._token = tok.he(token)  # type: ignore
```

Reply:
Why did you add the `# type: ignore` here?

Reply:
@thecode because of the linters.. the way the tok.py script works the linters can't decide which type is that and that causes an issue in the code tests.
I had to put this ignore.

Reply:
Your comment made me go over the `tok.py` and understand that we need to work on it. Please name it `token.py` and create a class in there

Reply:
@thecode this `tok.py` I got from the Switcher team. For us, it's a black box I rather not change anything in it.. I don't want to mess up with its logic..

Reply:
This is irrelevant, the code now goes into this repo, it should be aligned to this repo standards. You can (and should) first add test that cover the logic in this file, take few examples, run them before you modify and save the output, after that you can do you modifications and make sure the tests still pass. We need a test to check that the input token match the expected output anyhow else if something break in the future or someone changes something here we wouldn't know it.

Reply:
@thecode tests regarding tok.py added.
I tried change this tok.py file, any change done breaks it's logic and causes errors..
can't rename to token.py because its make confusion when calling to token.he() and "token" as variable

Reply:
I did some refactoring. you can review

Reply:
As mentioned earlier, please try to avoid refactoring during PR, you can do it later. I have no intention reviewing 700 lines over and over again. Only fix specific things which are requested in the PR

Reply:
@YogevBokobza This contribution does not comply with the project's license. We do not accept "black box" contributions. This is an open-source project; if the contribution is not open, it has no place in it.

Reply:
@thecode Nice!
I updated the tok.py based on that.